### PR TITLE
InstrumentationConfig part 3: HTTP headers and peer service mappings

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java [hunk 1]
```diff
@@ -19,7 +22,17 @@ public final class ArmeriaSingletons {
...
 
   static {
-    ArmeriaTelemetry telemetry = ArmeriaTelemetry.create(GlobalOpenTelemetry.get());
+    ArmeriaTelemetry telemetry =
+        ArmeriaTelemetry.builder(GlobalOpenTelemetry.get())
+            .setCapturedClientRequestHeaders(CommonConfig.get().getClientRequestHeaders())
+            .setCapturedClientResponseHeaders(CommonConfig.get().getClientResponseHeaders())
+            .setCapturedServerRequestHeaders(CommonConfig.get().getServerRequestHeaders())
+            .setCapturedServerResponseHeaders(CommonConfig.get().getServerResponseHeaders())
+            .addClientAttributeExtractor(
```
Comment (mateuszrzeszutek) [inline instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java:31]: This client/server method split is a bit awkward; I'm starting to think that there should be separate library instrumentation entry points for client/server, consumer/producer etc.

**Review by trask — COMMENTED** [review]
my recollection of old discussions is that `PeerServiceAttributesExtractor` was primarily for javaagent users, and that library users could just set `peer.service` directly in the (what's now) `TelemetryBuilder` since they would likely be creating a separate "interceptor" for each of their components that talk to different backends

do we still think this? and if so does that help us here if we move `PeerServiceAttributesExtractor` to javaagent only location?

File: instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java (hunk 1 above)
Comment (trask) [reply instrumentation/armeria-1.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/armeria/v1_3/ArmeriaSingletons.java:31]: this makes sense to me, I'd guess that most usage locations are dealing with client or server and not both(?)

**Comment by trask** [comment]
> my recollection of old discussions is that `PeerServiceAttributesExtractor` was primarily for javaagent users, and that library users could just set `peer.service` directly in the (what's now) `TelemetryBuilder` since they would likely be creating a separate "interceptor" for each of their components that talk to different backends
> 
> do we still think this? and if so does that help us here if we move `PeerServiceAttributesExtractor` to javaagent only location?

discussed in SIG meeting and realized there's really nowhere else really to put `PeerServiceAttributesExtractor`. The place it would go is javaagent-extension-api bootstrap, and looking at other classes in there it really doesn't fit.

**Review by trask — APPROVED** [review]

**Pull request merged**